### ISSUE TITLE
"systemctl preset-all" fails as "Failed to execute operation: Too many levels of symbolic links"

### ISSUE THREAD

**Issue by tereshonkov-roman** [systemd/systemd#2298]
For systemd 228.
"systemctl preset-all" fails as "Failed to execute operation: Too many levels of symbolic links"
if there is symlink 
"/etc/systemd/system/default.target -> /usr/lib/systemd/system/multi-user.target".

The same error comes during the very first boot. 

Any comments why preset fails?

**Comment by evverx** [comment]
```
# systemctl set-default multi-user.target
Removed symlink /etc/systemd/system/default.target.
Created symlink from /etc/systemd/system/default.target to /lib/systemd/system/multi-user.target.

# systemd-analyze set-log-level debug

# systemctl preset-all
Failed to execute operation: Too many levels of symbolic links

# journalctl -t systemd | grep default
...
Jan 11 14:50:24 systemd-testsuite systemd[1]: Preset file doesn't say anything about default.target, enabling.
...

# systemctl enable default.target
Failed to execute operation: Too many levels of symbolic links
```

See: [install: follow unit file symlinks in /usr, but not /etc when looking for [Install] data](https://github.com/poettering/systemd/commit/0ec0deaa30d0e68430f03fa6f32affa576481d18)

**Comment by tereshonkov-roman** [comment]
So, how to solve my problem then?

**Comment by keszybz** [comment]
Should be fixed by #2973 (3c6d8e5 ?).
